### ISSUE TITLE
Expression in `in` is broken in 7.1.4 and 8.0.0

### ISSUE THREAD

**Issue by blackst0ne** [rails/arel#531]
I am upgrading a rails 4.2 application to 5.0.    
I have some constructions based on arel which stopped working on arel `7.1.4` (rails `5.0.6`).

I have tested them on these versions: 
-  `6.0.4` (rails `4.2.10`) works.
-  `7.1.4` (rails `5.0.6`) doesn't work.
- `8.0.0` (rails `5.1.5`) doesn't work.
- `9.0.0` (rails `5.2.0`) works.

Code (simplified for readability) is this: 
 
```ruby
# Arel 6.0.4 and 9.0.0
> sql = Arel.sql(User.select(:id).to_sql)
=> "SELECT \"users\".\"id\" FROM \"users\""
> User.where(User.arel_table[:id].in(sql)).to_sql
=> "SELECT \"users\".* FROM \"users\" WHERE \"users\".\"id\" IN (SELECT \"users\".\"id\" FROM \"users\")"

# Arel 7.1.4 and 8.0.0
> sql = Arel.sql(User.select(:id).to_sql)
=> "SELECT \"users\".\"id\" FROM \"users\""
> User.where(User.arel_table[:id].in(sql)).to_sql
=> "SELECT \"users\".* FROM \"users\" WHERE \"users\".\"id\" IN (0)"
```

What can I do to get my code work again on `7.1.4` and `8.0.0` (as I'm going to upgrade to rails 5.1 later)?

**Comment by YorickPeterse** [comment]
For clarification: the Rails application in question is GitLab's Rails application.

**Comment by rafaelfranca** [comment]
First first step is trying to find which commit fixed it on 9.0. After that we can backport to the other versions.

**Comment by blackst0ne** [comment]
I checked locally. 
It seems `Arel` is not affected, everything is just fine. :thinking: 

Checking ActiveRecord...

**Comment by matthewd** [comment]
Per #523, Arel development is moving to [rails/rails](https://github.com/rails/rails).

If this issue is still relevant, please consider reopening it over there. (Note that the Rails repository does not accept feature request issues, and requires reproduction steps that rely on Active Record's documented API.)

**Comment by andsip** [comment]
> First first step is trying to find which commit fixed it on 9.0. After that we can backport to the other versions.

I think this was fixed in `9.0.0` via b416fca3.
